src/service/LocalStorageService.ts

```diff
@@ -0,0 +1,30 @@
+import TodoLine from "../model/TodoLine";
+
+export const LIST_KEY = "todoListKey"
+
+export default class LocalStorageService {
+
+    getData(): TodoLine[] {
+        const list = localStorage.getItem(LIST_KEY)
+        return list != null ? JSON.parse(list) as TodoLine[] : [] as TodoLine[]
+    }
+
+    addData(todo: TodoLine) {
+        const list = this.getData()
+        console.log(JSON.stringify(list))
+        list.push(todo)
+        localStorage.setItem(LIST_KEY, JSON.stringify(list))
+    }
```

Comment:
La fonction ne fonctionne pas si l'id existe deja

```suggestion
    addData(todo: TodoLine) {
    const list = this.getData();
    const currentIndex = list.findIndex((l) => l.id === todo.id);
    console.log(JSON.stringify(list));
    if (currentIndex >= 0) {
      list[currentIndex] = todo;
    } else {
      list.push(todo);
    }
    localStorage.setItem(LIST_KEY, JSON.stringify(list));
    }
```